sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -6671,6 +6719,100 @@ object functions {
   def to_json(e: Column): Column =
     to_json(e, Map.empty[String, String])
 
+  /**
+   * Masks the given string value. The function replaces characters with 'X' or 'x', and numbers
+   * with 'n'.
+   * This can be useful for creating copies of tables with sensitive information removed.
+   *
+   * @param input string value to mask. Supported types: STRING, VARCHAR, CHAR
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def mask(input: Column): Column = withExpr {
+    new Mask(input.expr)
+  }
+
+  // scalastyle:off line.size.limit
```

Comment:
ditto, we should avoid to use `scalastyle:off` for the new code

Reply:
got it. thanks.